cvat-core/src/annotations-actions.ts

```diff
@@ -0,0 +1,246 @@
+// Copyright (C) 2023 CVAT.ai Corporation
+//
+// SPDX-License-Identifier: MIT
+
+import { omit, throttle } from 'lodash';
+import { ArgumentError } from './exceptions';
+import { SerializedCollection } from './server-response-types';
+import { Job, Task } from './session';
+import { ObjectType } from './enums';
+import ObjectState from './object-state';
+import { getAnnotations, getCollection } from './annotations';
+
+export interface SingleFrameActionInput {
+    collection: Omit<SerializedCollection, 'tracks' | 'tags'>;
+    frameData: {
+        width: number;
+        height: number;
+        number: number;
+    };
+}
+
+export interface SingleFrameActionOutput {
+    collection: Omit<SerializedCollection, 'tracks' | 'tags'>;
+}
+
+export enum ActionParameterType {
+    SELECT = 'select',
+    NUMBER = 'number',
+}
+
+type ActionParameters = Record<string, {
+    type: ActionParameterType;
+    values: string[];
+    defaultValue: string;
+}>;
+
+export default class BaseSingleFrameAction {
+    /* eslint-disable @typescript-eslint/no-unused-vars */
+    public async init(
+        sessionInstance: Job | Task,
+        parameters: Record<string, string | number>,
+    ): Promise<void> {
+        throw new Error('Method not implemented');
+    }
+
+    public async destroy(): Promise<void> {
+        throw new Error('Method not implemented');
+    }
+
+    public async run(sessionInstance: Job | Task, input: SingleFrameActionInput): Promise<SingleFrameActionOutput> {
+        throw new Error('Method not implemented');
+    }
+
+    public get name(): string {
+        throw new Error('Method not implemented');
+    }
+
+    public get parameters(): ActionParameters | null {
+        throw new Error('Method not implemented');
+    }
+}
+
+class RemoveFilteredShapes extends BaseSingleFrameAction {
+    public async init(): Promise<void> {
+        // nothing to init
+    }
+
+    public async destroy(): Promise<void> {
+        // nothing to destroy
+    }
+
+    public async run(): Promise<SingleFrameActionOutput> {
+        return { collection: { shapes: [] } };
+    }
+
+    public get name(): string {
+        return 'Remove filtered shapes';
+    }
+
+    public get parameters(): ActionParameters | null {
+        return null;
```

Comment:
The reason why ``null`` instead of empty object is because:
- Boolean(null) === false
- Boolean({}) === true

It is just more convenient during coding
And IMHO null, visually looks better than empty object